environment.go

```diff
@@ -70,38 +189,20 @@ func (e *PipelineEnvironment) createTemplateParser() *template.Template {
 	return template.New("PipelineEnvironment").Funcs(fm)
 }
 
-func (e *PipelineEnvironment) ApplyTo(def *PipelineDefinition) error {
-	templateParser := e.createTemplateParser()
-	if err := e.applyToVolumes(def, templateParser); err != nil {
-		return err
-	}
-	return nil
-}
-
-func (e *PipelineEnvironment) applyToVolumes(def *PipelineDefinition, tp *template.Template) error {
-	pipelines, err := def.Pipelines()
+// ApplyTo executes the environment template parser on the provided data.
+func (e *PipelineEnvironment) ApplyTo(data []byte) ([]byte, error) {
```

Comment:
I would change the parameter name to something more meaningful e.g. `pipelineDefRaw`. Also one could think about having `ApplyTo` do the YAML parsing as well.

Reply:
If we decide to parse `.env.yml` files too this function would also work on them so `pipelineDefRaw` would not be optimal. Naming was inspired by the [encoding/json](https://golang.org/pkg/encoding/json/#pkg-index) package which operates on `data []byte`, I rename it to `rawFile` for now.